### PR TITLE
nagios: some refactoring

### PR DESCRIPTION
**Pull request by russoz**
##### SUMMARY
Minor refactoring in nagios module.

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
nagios

### CODE REVIEW

File: changelogs/fragments/5239-nagios-refactor.yaml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+  - nagios - Minor refactoring on parameter validation for different actions (https://github.com/ansible-collections/community.general/pull/5239).
```
Comment (felixfontein) [inline changelogs/fragments/5239-nagios-refactor.yaml:2]:
```suggestion
  - nagios - minor refactoring on parameter validation for different actions (https://github.com/ansible-collections/community.general/pull/5239).
```

File: plugins/modules/monitoring/nagios.py [hunk 2]
```diff
@@ -308,95 +304,44 @@ def main():
...
+        ),
+        required_if=[
+            ('action', 'downtime', ['host', 'services']),
+            ('action', 'delete_downtime', ['host', 'services']),
+            ('action', 'silence', ['host']),
+            ('action', 'unsilence', ['host']),
+            ('action', 'enable_alerts', ['host', 'services']),
+            ('action', 'disable_alerts', ['host', 'services']),
+            ('action', 'silence_nagios', ['services']),
+            ('action', 'unsilence_nagios', ['services']),
```
Comment (felixfontein) [inline plugins/modules/monitoring/nagios.py:326]: These do not seem to be required.
```suggestion
```

File: plugins/modules/monitoring/nagios.py [hunk 3]
```diff
@@ -308,95 +304,44 @@ def main():
...
+            ('action', 'command', ['services', 'command']),
```
Comment (felixfontein) [inline plugins/modules/monitoring/nagios.py:327]: Also here, `services` does not seem to be required:
```suggestion
            ('action', 'command', ['command']),
```

File: plugins/modules/monitoring/nagios.py (hunk 2 above)
Comment (russoz) [reply plugins/modules/monitoring/nagios.py:326]: Thanks for spotting!

**Review by felixfontein — APPROVED** [review]

**Pull request merged**